src/components/common/Map/NewTarget/index.js

```diff
@@ -0,0 +1,23 @@
+import { useHistory } from 'react-router-dom';
+import { useMapEvents } from 'react-leaflet';
+import { setNewTargetCoords } from '../../../../state/actions/targetActions';
+import useDispatch from '../../../../hooks/useDispatch';
+
+const NewTarget = ({ setNewMarker }) => {
+  const history = useHistory();
+  const updateCoordinates = useDispatch(setNewTargetCoords);
+
+  useMapEvents({
+    click(e) {
+      const coordinates = [e.latlng.lat, e.latlng.lng];
+      setNewMarker(coordinates);
+      updateCoordinates(coordinates);
+      history.push('/'); // forces re-render of create target form, to set new coords on each click on the map
+      history.push('/targets/new');
+    }
+  });
+
+  return null;
```

Comment:
Can you just remove the return line or you need the `null` value?

Reply:
Same as with UpdateCenter component 😄